### PR TITLE
Change logout from sync to async due to SDK changes

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Change logout fn from sync to async to match [changes to the SDK](https://github.com/AudiusProject/audius-protocol/pull/3420/files) (SDK logout changed to async due to Hedgehog change that accommodates using a custom local storage interface).

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?
Need to link hedgehog + libs locally and test in browser; will update this when done

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/common/store/sign-out/sagas.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ function* watchSignOut() {
     if (NATIVE_MOBILE) {
       disablePushNotifications()
       yield put(make(Name.SETTINGS_LOG_OUT, {}))
-      signOut()
+      call(signOut)
```
Comment (sliptype) [inline packages/web/src/common/store/sign-out/sagas.ts:17]: Do we need a `yield` here?

**Review by sliptype — APPROVED** [review]
Awesome, making progress 🎉

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--hedgehog-local-storage

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--hedgehog-local-storage

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--hedgehog-local-storage

**Pull request merged**